src/Interpreters/AsynchronousInsertQueue.cpp

```diff
@@ -176,11 +179,21 @@ AsynchronousInsertQueue::AsynchronousInsertQueue(ContextPtr context_, size_t poo
     , pool_size(pool_size_)
     , flush_on_shutdown(flush_on_shutdown_)
     , queue_shards(pool_size)
-    , pool(CurrentMetrics::AsynchronousInsertThreads, CurrentMetrics::AsynchronousInsertThreadsActive, CurrentMetrics::AsynchronousInsertThreadsScheduled, pool_size)
+    , flush_time_history_per_queue_shard(pool_size)
+    , pool(
+          CurrentMetrics::AsynchronousInsertThreads,
+          CurrentMetrics::AsynchronousInsertThreadsActive,
+          CurrentMetrics::AsynchronousInsertThreadsScheduled,
+          pool_size)
 {
     if (!pool_size)
         throw Exception(ErrorCodes::BAD_ARGUMENTS, "pool_size cannot be zero");
 
+    const auto & settings = getContext()->getSettingsRef();
+
+    for (size_t i = 0; i < pool_size; ++i)
+        queue_shards[i].busy_timeout_ms = settings.async_insert_busy_timeout_ms;
```

Comment:
I've been considering a smooth transition from the state 'enable_async_insert_adaptive_busy_timeout=0' to '1'. async_insert_busy_timeout_ms is a timeout anticipated by a customer, and the adaptive algorithm treats it as the initial point. It minimizes the value if inserts are not frequent.

Reply:
> is a timeout anticipated by a customer

if they touch for some reason the timeout setting, otherwise they don't anticipate anything as long as it works

the problem here is that we would like to solve the timeout of the initial insert, and if the user set `async_insert_busy_timeout_ms` to some high value, it will take a long time for even a simple insert of 1 row.
IMO we should start from minimal flushtimeout and gradually grow from there if the insert rate increases.